beacon-chain/db/kv/wss.go

```diff
@@ -0,0 +1,89 @@
+package kv
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	statev2 "github.com/prysmaticlabs/prysm/beacon-chain/state/v2"
+	"github.com/prysmaticlabs/prysm/config/params"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/wrapper"
+)
+
+// SaveOrigin loads an ssz serialized Block & BeaconState from an io.Reader
+// (ex: an open file) prepares the database so that the beacon node can begin
+// syncing, using the provided values as their point of origin. This is an alternative
+// to syncing from genesis, and should only be run on an empty database.
+func (s *Store) SaveOrigin(ctx context.Context, stateReader, blockReader io.Reader) error {
```

Comment:
Unit test?

Reply:
as discussed in discord this is an unused code path in this branch, but i'm leaving it to avoid annoying git conflicts. will be adding more tests in the branch where checkpoint sync flags are introduced.